File: java/com/example/spottio/CommentAdapter.java
package com.example.spottio;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

public class CommentAdapter extends RecyclerView.Adapter<CommentAdapter.CommentViewHolder> {

    private List<Comment> commentList;

    public CommentAdapter(List<Comment> commentList) {
        this.commentList = commentList;
    }

    @NonNull
    @Override
    public CommentViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_comment, parent, false);
        return new CommentViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CommentViewHolder holder, int position) {
        Comment comment = commentList.get(position);

        holder.tvUser.setText(comment.getAuthor());
        holder.tvText.setText(comment.getText());
        holder.tvDate.setText(comment.getFormattedDate());
    }

    @Override
    public int getItemCount() {
        return (commentList != null) ? commentList.size() : 0;
    }

    static class CommentViewHolder extends RecyclerView.ViewHolder {
        TextView tvUser, tvText, tvDate;

        public CommentViewHolder(@NonNull View itemView) {
            super(itemView);
            tvUser = itemView.findViewById(R.id.tvCommentUser);
            tvText = itemView.findViewById(R.id.tvCommentText);
            tvDate = itemView.findViewById(R.id.tvCommentDate);
        }
    }
}

File: app/src/main/java/com/example/spottio/CommentAdapter.kt
package com.example.spottio

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.spottio.CommentAdapter.CommentViewHolder

class CommentAdapter(private val commentList: List<Comment>?) :
    RecyclerView.Adapter<CommentViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CommentViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_comment, parent, false)
        return CommentViewHolder(view)
    }

    override fun onBindViewHolder(holder: CommentViewHolder, position: Int) {
        val comment = commentList!![position]

        holder.tvUser.text = comment.author
        holder.tvText.text = comment.text
        holder.tvDate.text = comment.formattedDate
    }

    override fun getItemCount(): Int {
        return if ((commentList != null)) commentList.size else 0
    }

    class CommentViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvUser: TextView = itemView.findViewById(R.id.tvCommentUser)
        var tvText: TextView = itemView.findViewById(R.id.tvCommentText)
        var tvDate: TextView = itemView.findViewById(R.id.tvCommentDate)
    }
}